nodes/src/nodes/preprocessor_native/src/chunker.cpp

```
@@ -0,0 +1,344 @@
+            }
+            sent_start = next_start;
+        } else {
+            sent_start = sent_end;
+        }
+    }
+
+    return chunks;
+}
+
+// Split text by a separator, keeping the separator at the end of each piece (like LangChain keep_separator=true)
+static std::vector<std::string_view> split_by(std::string_view text, std::string_view sep) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)